### PR TITLE
AutoComplete : ReadData

### LINKED ISSUE #3450: Autocomplete : Load data on demand (Similar to ReadData on DataGrid) : Feature
Autocomplete : Load data on demand (Similar to ReadData on DataGrid) : Feature

Requested on #410

### PR DESCRIPTION
**Pull request by David-Moreira**
Closes #3450

### CODE REVIEW

**Comment by David-Moreira** [comment]
(Ongoing)

File: Source/Extensions/Blazorise.Components/Autocomplete.razor.cs [hunk 1]
```diff
@@ -181,6 +232,8 @@ protected async Task OnTextChangedHandler( string text )
             await SearchChanged.InvokeAsync( CurrentSearch );
             await SelectedTextChanged.InvokeAsync( SelectedText );
 
+            await HandleReadData();
```
Comment (stsrki) [inline Source/Extensions/Blazorise.Components/Autocomplete.razor.cs:235]: Should this be behind the `if ( ManualReadMode )`?

File: Source/Extensions/Blazorise.Components/Autocomplete.razor.cs [hunk 2]
```diff
@@ -65,10 +70,54 @@ public partial class Autocomplete<TItem, TValue> : BaseAfterRenderComponent, ICl
...
+        }
+
+        /// <summary>
+        /// Triggers the reload of the <see cref="Autocomplete{TItem, TValue}"/> data.
+        /// Makes sure not to reload if the <see cref="Autocomplete{TItem, TValue}"/> is in a loading state.
+        /// </summary>
+        /// <returns>Returns the awaitable task.</returns>
+        public async Task Reload( CancellationToken cancellationToken = default )
+        {
+            if ( IsLoading )
```
Comment (stsrki) [inline Source/Extensions/Blazorise.Components/Autocomplete.razor.cs:106]: Usually, we don't have an Is prefix.

Comment (stsrki) [inline Source/Extensions/Blazorise.Components/Autocomplete.razor.cs:106]: Usually, we don't have an Is prefix.

Comment (David-Moreira) [reply Source/Extensions/Blazorise.Components/Autocomplete.razor.cs:106]: Based on DataGrid :) We can fix both then.

File: Source/Extensions/Blazorise.Components/Autocomplete.razor.cs [hunk 3]
```diff
@@ -25,6 +25,11 @@ public partial class Autocomplete<TItem, TValue> : BaseAfterRenderComponent, ICl
     {
         #region Members
 
+        /// <summary>
+        /// Returns true if ReadData will be invoked.
+        /// </summary>
+        protected bool IsLoading { get; set; }
```
Comment (stsrki) [inline Source/Extensions/Blazorise.Components/Autocomplete.razor.cs:31]: Move down in file under Properties regien

File: Source/Extensions/Blazorise.Components/Autocomplete.razor.cs (hunk 2 above)
Comment (David-Moreira) [reply Source/Extensions/Blazorise.Components/Autocomplete.razor.cs]: Well I was just starting a PR to fix DataGrid... And there's a bunch of them there prefix 'Is'.  Most of them are protected, so the impact wouldn't be big, but we might have public one's....
Leave it for 2.0? Can I open an issue with this kind of cleanup?
image

I do have an habbit of using the prefix `Is`...So I guess the majority might have been introduced by me....
But even utilities we may have that pattern... And I think you're the author of these?
image

So finish the document for Blazorise conventions first, and leave it for 2.0 (on 2.0 we try to make sure we comply to our own defined conventions)?

**Comment by David-Moreira** [comment]
Closes #3865

File: Source/Extensions/Blazorise.Components/Autocomplete.razor.cs (hunk 2 above)
Comment (stsrki) [reply Source/Extensions/Blazorise.Components/Autocomplete.razor.cs]: Property getter are allowed to have prefix, see https://github.com/Megabit/Blazorise/issues/3424#issuecomment-1146800048

Comment (David-Moreira) [reply Source/Extensions/Blazorise.Components/Autocomplete.razor.cs]: I see.

**Comment by stsrki** [comment]
Is this ready for final review?

**Comment by David-Moreira** [comment]
yes

**Review by stsrki — APPROVED** [review]
LGTM

**Pull request merged**